task/07-yield-tasks.js

```diff
@@ -82,7 +95,36 @@ function* getFibonacciSequence() {
  *
  */
 function* depthTraversalTree(root) {
-    throw new Error('Not implemented');
+    if(!root.children) {        
+        return root;
+    }
+
+    let stack = [];
+    let index = 0;    
+	while(index >= 0) {
+		if(stack.length == 0 || stack[stack.length - 1].n < root.n) {
+            yield root;
+            if(root.children) {
+                root.nextChildToHandle = 0;
+            }
+            stack.push(root);		    
+        }
+        
+		if(stack[index].children) {
```

Comment:
it is necessary to observe indents between operators, brackets and indents for code lines.
https://javascript.info/coding-style